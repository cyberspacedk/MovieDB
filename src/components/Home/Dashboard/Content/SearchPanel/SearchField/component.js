import React from 'react';
import PropTypes from 'prop-types';
import { Input, Form } from 'antd';

const { Item } = Form;
const { Search } = Input;

const SearchField = ({
  field,
  form: { touched, errors, handleSubmit },
  ...props
}) => (
  <Item
    validateStatus={touched[field.name] && errors[field.name] && 'error'}
    help={errors[field.name]}
  >
    <Search
      onSubmit={handleSubmit}
      placeholder="Enter movie name"
      size="large"
      enterButton="Search"
      className="top-margin"
      onSearch={handleSubmit}
      {...field}
      {...props}
    />
  </Item>
);

SearchField.propTypes = {
  field: PropTypes.shape({ name: PropTypes.string }),
  form: PropTypes.shape({
    touched: PropTypes.object,
    errors: PropTypes.object,
  }),
};

export default SearchField;
