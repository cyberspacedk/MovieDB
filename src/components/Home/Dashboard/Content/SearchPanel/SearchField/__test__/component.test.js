import React from 'react';
import { shallow } from 'enzyme';
import SearchField from '../component';

describe('Component: SearchField', () => {
  const props = {
    field: {
      name: 'search',
    },
    form: {
      errors: {},
      touched: {},
    },
  };
  const container = shallow(<SearchField {...props} />);

  it('Snapshot: should match. Field without errors', () => {
    expect(container).toMatchSnapshot();
  });

  it('Snapshot: should match. Field with errors', () => {
    const nextProps = {
      ...props,
      form: {
        errors: {
          search: 'Some error',
        },
        touched: {
          search: 'required field',
        },
      },
    };
    const container = shallow(<SearchField {...nextProps} />);

    expect(container).toMatchSnapshot();
  });
});
