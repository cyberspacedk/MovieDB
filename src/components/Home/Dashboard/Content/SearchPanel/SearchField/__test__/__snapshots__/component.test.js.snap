// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SearchField Snapshot: should match. Field with errors 1`] = `
<FormItem
  hasFeedback={false}
  help="Some error"
  validateStatus="error"
>
  <Search
    className="top-margin"
    enterButton="Search"
    name="search"
    placeholder="Enter movie name"
    size="large"
  />
</FormItem>
`;

exports[`Component: SearchField Snapshot: should match. Field without errors 1`] = `
<FormItem
  hasFeedback={false}
>
  <Search
    className="top-margin"
    enterButton="Search"
    name="search"
    placeholder="Enter movie name"
    size="large"
  />
</FormItem>
`;
